sources/org.osbuild.dnf

```diff
@@ -4,20 +4,47 @@ import json
 import sys
 
 
-def main(options, sources):
+def read_file(path):
+    with open(path) as f:
+        return f.read()
+
+
+def main(options, sources, secrets):
     repos = options.get("repos", {})
+    repo_secrets = secrets.get("repos", {})
 
     reply = []
-    for s in sources:
+    for checksum in sources:
         try:
-            repo = {
-                "checksum": s,
-                **repos[s]
-            }
+            source_repo = repos[checksum]
+            source_repo_secrets = repo_secrets.get(checksum, {})
         except KeyError:
-            json.dump({"error": f"source unknown: {s}"}, sys.stdout)
+            json.dump({"error": f"source unknown: {checksum}"}, sys.stdout)
             return 1
 
+        repo = {"checksum": checksum}
+
+        if "baseurl" in source_repo:
+            repo["baseurl"] = source_repo["baseurl"]
+        elif "mirrorlist" in source_repo:
+            repo["mirrorlist"] = source_repo["mirrorlist"]
+        elif "metalink" in source_repo:
+            repo["metalink"] = source_repo["metalink"]
+        else:
+            json.dump({"error": f"repo {checksum} is missing baseurl, mirrorlist, or metalink key"}, sys.stdout)
+
+        if "sslcacert" in source_repo:
```

Comment:
I wonder if we should allow files here rather than requiring the information to be inline? I know I argued for this way in the past, but I think I was wrong ;) I think it isn't right to implicitly tie the osbuild to the host in this way, it should at least be explicit. And in the case of composer, we probably want to read these files either from the composer host (rather than the worker host) or require the user to pass them in to the API.

Reply:
Agreed. Felt dirty, but so convenient.

For the record, I've found a quick way to convert a file into a properly escaped JSON string:
```
cat /etc/rhsm/ca/redhat-uep.pem | jq --slurp --raw-input .
```